### ISSUE TITLE
Can't use autofix to remove some code

### ISSUE THREAD

**Issue by aryx** [semgrep/semgrep#6318]
with this rule:
```
 - id: python-typing-osemgrep
   pattern: from typing import $X
   fix: ""
   languages: [ python ]
   message: found one
   severity: ERROR
```
I can't use it on some codebase to remove all the typing imports.
`semgrep --config ~/refactoring.yaml . --autofix` does not apply any fixes.

**Comment by r2c-demo** [comment]
This issue is synced in Linear at https://linear.app/r2c/issue/PA-2000/cant-autofix-to-remove-some-code. _Note_: this link is for r2c use only and is not accessible publicly.

**Comment by alanisaac** [comment]
I'm also looking for the same feature.  I think the thread in this issue is related: https://github.com/returntocorp/semgrep/issues/5674

**Comment by joshlory** [comment]
Without this it's hard to use Semgrep for code cleanup. @nmote I think the line you're looking for is here?

https://github.com/returntocorp/semgrep/blob/33d49d5d351601598c882eb3fa9f758c441ea101/cli/src/semgrep/core_output.py#L168

> I don't think AST-based autofix would affect it. It doesn't affect fix-regex at all, since fix-regex is by definition text-based. I bet this is due to a Python truthiness test somewhere like `if replacement:`, since the empty string is falsy. I bet that finding that spot and replacing `if replacement:` with `if replacement is not None:` would address this issue.

https://github.com/returntocorp/semgrep/issues/5674#issuecomment-1307473104

**Comment by nmote** [comment]
@joshlory that looks pretty likely. Are you interested in putting up a PR?

**Comment by nmote** [comment]
Fixed by #7772
